java/client/src/main/java/glide/connectors/handlers/ChannelHandler.java

```diff
@@ -68,12 +68,13 @@ public ChannelHandler(
     /**
      * Complete a protobuf message and write it to the channel (to UDS).
      *
-     * @param request Incomplete request, function completes it by setting callback ID
+     * @param requestWithoutCallback Incomplete request, function completes it by setting callback ID
      * @param flush True to flush immediately
      * @return A response promise
      */
-    public CompletableFuture<Response> write(RedisRequest.Builder request, boolean flush) {
+    public CompletableFuture<Response> write(RedisRequest requestWithoutCallback, boolean flush) {
         var commandId = callbackDispatcher.registerRequest();
+        var request = RedisRequest.newBuilder(requestWithoutCallback);
```

Comment:
what about performance of this step? I afraid that can't afford creating new builder - that is the reason why initial implementation operated with incomplete builder instead of a request.

Reply:
Passing a builder around isn't great.  The builder isn't comparable, so it's hard to test.  
I can revert back, and try something else if you're concerned about performance.